internal/services/network/custom_ip_prefix_resource.go

```diff
@@ -0,0 +1,550 @@
+}
+
+var (
+	_ sdk.Resource           = CustomIpPrefixResource{}
+	_ sdk.ResourceWithUpdate = CustomIpPrefixResource{}
+)
+
+type CustomIpPrefixResource struct {
+	client *network.CustomIPPrefixesClient
+}
+
+func (CustomIpPrefixResource) ResourceType() string {
+	return "azurerm_custom_ip_prefix"
+}
+
+func (CustomIpPrefixResource) ModelObject() interface{} {
+	return &CustomIpPrefixModel{}
+}
+
+func (CustomIpPrefixResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.CustomIpPrefixID
+}
+func (r CustomIpPrefixResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.CustomIpPrefixName,
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"cidr": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: func(i interface{}, k string) (warnings []string, errors []error) {
+				v, ok := i.(string)
+				if !ok {
+					errors = append(errors, fmt.Errorf("expected type of %s to be string", k))
+					return
+				}
+
+				if _, _, err := net.ParseCIDR(v); err != nil {
+					errors = append(errors, fmt.Errorf("expected %q to be a valid IPv4 or IPv6 network, got %v: %v", k, i, err))
+				}
+
+				return
+			},
+		},
+
+		"roa_validity_end_date": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: func(i interface{}, k string) (warnings []string, errors []error) {
+				v, ok := i.(string)
+				if !ok {
+					errors = append(errors, fmt.Errorf("expected type of %q to be string", k))
+					return warnings, errors
+				}
+
+				if _, err := time.Parse("2006-01-02", v); err != nil {
+					errors = append(errors, fmt.Errorf("expected %q to be a valid date in the format YYYY-MM-DD, got %q: %+v", k, i, err))
+				}
+
+				return warnings, errors
+			},
+		},
+
+		"wan_validation_signed_message": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"commissioning_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"internet_advertising_disabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"tags": commonschema.Tags(),
+
+		"zones": commonschema.ZonesMultipleOptionalForceNew(),
+	}
+}
+
+func (r CustomIpPrefixResource) Attributes() map[string]*pluginsdk.Schema {
+	return nil
+}
+
+func (r CustomIpPrefixResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 6 * time.Hour,
+
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			r.client = metadata.Client.Network.CustomIPPrefixesClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			timeout, _ := ctx.Deadline()
```

Comment:
We should probably do this after the `CreateOrUpdate` call to be safe as that call can take some time to return the future iirc? We should probably also not discard the error in case there's a problem with the context for some reason?